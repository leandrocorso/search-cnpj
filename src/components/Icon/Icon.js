import React from 'react';
import iconPackage from './package.svg';

const Icon = ({ src, ...rest }) => (
  <svg data-test="icon" src={src} {...rest}>
    <use xlinkHref={`${iconPackage}#icon-${src}`} />
  </svg>
);

export default Icon;
