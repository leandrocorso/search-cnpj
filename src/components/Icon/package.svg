<svg aria-hidden="true" style="position: absolute; width: 0; height: 0; overflow: hidden;" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<defs>
<symbol id="icon-checkmark" viewBox="0 0 32 32">
<title>checkmark</title>
<path d="M27 4l-15 15-7-7-5 5 12 12 20-20z"></path>
</symbol>
<symbol id="icon-cross" viewBox="0 0 32 32">
<title>cross</title>
<path d="M31.708 25.708c-0-0-0-0-0-0l-9.708-9.708 9.708-9.708c0-0 0-0 0-0 0.105-0.105 0.18-0.227 0.229-0.357 0.133-0.356 0.057-0.771-0.229-1.057l-4.586-4.586c-0.286-0.286-0.702-0.361-1.057-0.229-0.13 0.048-0.252 0.124-0.357 0.228 0 0-0 0-0 0l-9.708 9.708-9.708-9.708c-0-0-0-0-0-0-0.105-0.104-0.227-0.18-0.357-0.228-0.356-0.133-0.771-0.057-1.057 0.229l-4.586 4.586c-0.286 0.286-0.361 0.702-0.229 1.057 0.049 0.13 0.124 0.252 0.229 0.357 0 0 0 0 0 0l9.708 9.708-9.708 9.708c-0 0-0 0-0 0-0.104 0.105-0.18 0.227-0.229 0.357-0.133 0.355-0.057 0.771 0.229 1.057l4.586 4.586c0.286 0.286 0.702 0.361 1.057 0.229 0.13-0.049 0.252-0.124 0.357-0.229 0-0 0-0 0-0l9.708-9.708 9.708 9.708c0 0 0 0 0 0 0.105 0.105 0.227 0.18 0.357 0.229 0.356 0.133 0.771 0.057 1.057-0.229l4.586-4.586c0.286-0.286 0.362-0.702 0.229-1.057-0.049-0.13-0.124-0.252-0.229-0.357z"></path>
</symbol>
<symbol id="icon-arrow-right2" viewBox="0 0 32 32">
<title>arrow-right2</title>
<path d="M19.414 27.414l10-10c0.781-0.781 0.781-2.047 0-2.828l-10-10c-0.781-0.781-2.047-0.781-2.828 0s-0.781 2.047 0 2.828l6.586 6.586h-19.172c-1.105 0-2 0.895-2 2s0.895 2 2 2h19.172l-6.586 6.586c-0.39 0.39-0.586 0.902-0.586 1.414s0.195 1.024 0.586 1.414c0.781 0.781 2.047 0.781 2.828 0z"></path>
</symbol>
<symbol id="icon-line-chart" viewBox="0 0 32 28">
<title>line-chart</title>
<path d="M32 24v2h-32v-24h2v22h30zM30 4.5v6.797c0 0.438-0.531 0.672-0.859 0.344l-1.891-1.891-9.891 9.891c-0.203 0.203-0.516 0.203-0.719 0l-3.641-3.641-6.5 6.5-3-3 9.141-9.141c0.203-0.203 0.516-0.203 0.719 0l3.641 3.641 7.25-7.25-1.891-1.891c-0.328-0.328-0.094-0.859 0.344-0.859h6.797c0.281 0 0.5 0.219 0.5 0.5z"></path>
</symbol>
</defs>
</svg>
