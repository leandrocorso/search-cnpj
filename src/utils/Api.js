import axios from 'axios';

import { consts, Auth } from './';

const instance = axios.create({
	baseURL: consts.BASE_URL,
});

if (Auth.getToken()) {
	instance.defaults.headers.common[
		'Authorization'
	] = `Bearer ${Auth.getToken()}`;
}

instance.interceptors.request.use(
	function(config) {
		if (config.headers.common.Authorization) {
			if (!Auth.isLogged()) {
				throw new axios.Cancel('Invalid token');
			}
		}

		return config;
	},
	function(error) {
		return Promise.reject(error);
	}
);

class Api {
	static get(uri) {
		return instance.get(`${uri}`);
	}
}

export default Api;
