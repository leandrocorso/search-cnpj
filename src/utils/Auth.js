import Cookies from 'universal-cookie';

import { consts } from './';

const cookies = new Cookies();

class Auth {

    static getToken = () => ( cookies.get('token') );
    
    static setToken = () => ( cookies.set('token', consts.TOKEN, { path: '/' } ) );

    static isLogged = () => {
        if (!cookies.get('token')) return false;
        return cookies.get('token') === consts.TOKEN;
    };

}

export default Auth;