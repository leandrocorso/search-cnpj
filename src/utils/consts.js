const BASE_URL = 'http://jsonplaceholder.typicode.com';
const TOKEN = '23456789';
const VALID_CNPJ = '12345678000123';

export default { BASE_URL, TOKEN, VALID_CNPJ };
