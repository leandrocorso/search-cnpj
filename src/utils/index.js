import consts from './consts';
import Auth from './Auth';
import Api from './Api';

const findByTestAttr = (component, attr) => (
    component.find(`[data-test="${attr}"]`)
);

const int = (string) => {
    return string.replace(/[^0-9]/g,'');
}


export { consts, Auth, Api, findByTestAttr, int };
