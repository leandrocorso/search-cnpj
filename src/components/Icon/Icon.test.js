import React from 'react';
import { shallow } from 'enzyme';

import { findByTestAttr } from '../../utils';

import Icon from './Icon';

const setup = (props = {}) => (
    shallow(<Icon { ...props } />)
);

let component;

describe('Icon component', () => {

    beforeEach(() => { 
        component = setup();
    });

    test('Should render the component correctly', () => {
        expect(component).toMatchSnapshot();
    });
    
    test('Should have a source', () => {
        const wrapper = findByTestAttr(component, 'icon');
        expect(wrapper).toHaveProp('src');
    });

})