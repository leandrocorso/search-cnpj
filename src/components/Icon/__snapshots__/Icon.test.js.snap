// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Should render the component correctly 1`] = `
<svg
  data-test="icon"
>
  <use
    xlinkHref="package.svg#icon-undefined"
  />
</svg>
`;
